scrapylib/hcf.py

```diff
@@ -156,7 +159,13 @@ def process_spider_output(self, response, result, spider):
                 if request.meta.get('use_hcf', False):
                     if request.method == 'GET':  # XXX: Only GET support for now.
                         slot = slot_callback(request)
-                        if not request.url in self.new_links[slot]:
+                        filtered = False
+                        if not request.meta.get('hcf_dont_filter', False):
+                            if request.url not in self.new_links[slot]:
+                                self.new_links[slot].add(request.url)
+                            else:
+                                filtered = True
+                        if not filtered:
```

Comment:
> indenting everything inside the else

This is not correct as you've said later, "It is the inverse"

> It is the inverse so you can probably use continue since it is inside a loop

Ah that's true and I've actually considered this style of using a `continue` :) The reason why I didn't use the `continue` is it's already insider 4 levels of nested `if`s in the loop, too deep I think to make it clear. Also I think other kinds of filtering might be introduced in the future. :)